src/scripts/main.js

```diff
@@ -1,3 +1,279 @@
+  employeeForm.append(button);
+
+  const firstScript = document.querySelector('body script');
+
+  document.body.insertBefore(employeeForm, firstScript);
+}
+
+newEmployeeForm();
+
+const form = document.querySelector('.new-employee-form');
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const COUNT_OF_COLUMNS = 5;
+
+  const newEmloyee = {
+    employeeName: form.querySelector('input[data-qa="name"]').value.trim(),
+    position: form.querySelector('input[data-qa="position"]').value.trim(),
+    office: form.querySelector('select[data-qa="office"]').value,
+    age: form.querySelector('input[data-qa="age"]').value.trim(),
+    salary: +form.querySelector('input[data-qa="salary"]').value.trim(),
+  };
+
+  if (!validateEmployee(newEmloyee)) {
+    return;
+  }
+
+  const tbody = pageTable.querySelector('tbody');
+
+  const newTr = document.createElement('tr');
+
+  for (let i = 0; i < COUNT_OF_COLUMNS; i++) {
+    const newTd = document.createElement('td');
+
+    if (i === 4) {
+      newTd.innerText = convertToCurrency(
+        newEmloyee[Object.keys(newEmloyee)[i]],
+      );
+    } else {
+      newTd.innerText = newEmloyee[Object.keys(newEmloyee)[i]];
+    }
+
+    newTr.append(newTd);
+  }
+
+  tbody.append(newTr);
+
+  form.reset();
+});
+
+pageTable.addEventListener('click', (e) => {
+  const tbody = pageTable.querySelector('tbody');
+  const rows = [...tbody.querySelectorAll('tr')];
+  const th = e.target.closest('th');
+  const tr = e.target.closest('tr');
+
+  if (th) {
+    const thIndex = th.cellIndex;
+
+    if (pageTableState.sortedColumn !== thIndex) {
+      pageTableState.sortedBy = null;
+      pageTableState.sortedColumn = thIndex;
+    }
+
+    const isAscending =
+      !pageTableState.sortedBy || pageTableState.sortedBy === 'DESC';
+
+    pageTableState.sortedBy = isAscending ? 'ASC' : 'DESC';
+
+    const sortedRows = rows.sort((rowA, rowB) => {
+      const a = rowA.children[th.cellIndex].textContent;
+      const b = rowB.children[th.cellIndex].textContent;
+
+      const isCurrency = (val) =>
+        '$€£¥CHF₹₽₩R₺RM฿Rp₱ر.سد.إ₪E£₫₦₨৳'.includes(val[0]) &&
+        '0123456789'.includes(val[1]);
+
+      const compareValues = (valA, valB) => {
+        if (isCurrency(valA)) {
+          return extractAmount(valA) - extractAmount(valB);
+        }
+
+        return valA.localeCompare(valB);
+      };
+
+      return isAscending ? compareValues(a, b) : compareValues(b, a);
+    });
+
+    tbody.append(...sortedRows);
+  }
+
+  if (tr && tr.closest('tbody') === tbody) {
+    const trList = [...tbody.querySelectorAll('tr')];
+
+    pageTableState.selectedRow = tr;
+
+    trList.forEach((row) => {
+      row.classList.toggle('active', row === pageTableState.selectedRow);
+    });
+  }
+});
+
+function saveChanges(input, initialValue) {
+  const newValue = input.value.trim() || initialValue;
+  const cell = input.parentElement;
+
+  cell.textContent = newValue;
```

Comment:
for salary would be nice to transform it in the same way as you do when adding a new row